### ISSUE TITLE
slow processing of long lines

### ISSUE THREAD

**Issue by sfc-gh-pmardziel** [mondeja/mkdocs-include-markdown-plugin#221]
When processing a file that has a long line (no line breaks), the plugin takes more time than expected. This becomes especially problematic when content is markdown that includes an embedded image whose long base64 encoding is on a single line. Here is a small example using ipsum lorem: https://gist.github.com/sfc-gh-pmardziel/15c1242a9e4e3760ba15a0d324d414da . This example takes about 50 seconds to process the computation that is indicated by this `--verbose` logline:

```
DEBUG   -  Running `page_markdown` event from plugin 'include-markdown'
```

Note that in this example and in base64 encodings, the default `opening_tag` and `closing_tag` are not found anywhere in the line and changing them to something that is longer/harder to match has no affect on the slowness. I thus theorize that some other aspect of the regex that is being matched is responsible for the long run time. 

# Environment:
```
> pip list | grep mkdocs
mkdocs                            1.6.0
mkdocs-autorefs                   1.0.1
mkdocs-get-deps                   0.2.0
mkdocs-include-markdown-plugin    6.2.0
mkdocs-jupyter                    0.24.7
mkdocs-material                   9.5.26
mkdocs-material-extensions        1.3.1
mkdocs-redirects                  1.2.1
mkdocstrings                      0.25.1
mkdocstrings-python               1.10.3
```

```
> python --version
Python 3.12.3
```

```
> uname -a
Darwin FVWKKW4JLF 23.5.0 Darwin Kernel Version 23.5.0: Wed May  1 20:17:33 PDT 2024; root:xnu-10063.121.3~5/RELEASE_ARM64_T6031 arm64
```
This is an "Apple M1" mac in case it wasn't clear from that uname.

**Comment by mondeja** [comment]
As a workaround, with v6.2.0 you can include the image in your markdown file using `recursive=false` and the regex will not be applied to that long line.

**Comment by mondeja** [comment]
Fixed on [v6.2.1](https://github.com/mondeja/mkdocs-include-markdown-plugin/releases/tag/v6.2.1)
